specification/trace/semantic_conventions/messaging.md

```diff
@@ -354,15 +337,16 @@ Attributes in the `messaging.message` or `messaging.{system}.message` namespace
 
 For batch operations, per-message attributes are usually different and cannot be set on the corresponding span. In such cases the attributes MAY be set on links. See [Batch Receiving](#batch-receiving) and [Batch Processing](#batch-processing) for more information on correlation using links.
 
-Some messaging systems (e.g., Kafka, Azure EventGrid) allow publishing a single batch of messages to different topics. In such cases, the attributes in `messaging.destination` and `messaging.source` MAY be
-set on links. Instrumentations MAY set source and destination attributes on the span if all messages in the batch share the same destination or source.
+Some messaging systems (e.g., Kafka, Azure EventGrid) allow publishing a single batch of messages to different topics. In such cases, the attributes in `messaging.destination` and `messaging.destination_original` MAY be
+set on links. Instrumentations MAY set destination attributes on the span if all messages in the batch share the same destination.
 
 ### Attributes specific to certain messaging systems
 
-All attributes that are specific for a messaging system SHOULD be populated in `messaging.{system}` namespace. Attributes that describe a message, a destination, a source, a consumer or a batch of messages SHOULD be populated under the corresponding namespace:
+All attributes that are specific for a messaging system SHOULD be populated in `messaging.{system}` namespace. Attributes that describe a message, a destination, a consumer, or a batch of messages SHOULD be populated under the corresponding namespace:
 
 * `messaging.{system}.message`: Describes attributes for individual messages
-* `messaging.{system}.destination` and `messaging.{system}.source`: Describe the destination and source a message (or a batch) are published to and received from respectively. The combination of attributes in these namespaces should uniquely identify the entity and include properties significant for this messaging system. For example, Kafka instrumentations should include partition identifier.
+* `messaging.{system}.destination`: Describe the destination a message (or a batch) are published to and received from respectively. The combination of attributes in these namespaces should uniquely identify the entity and include properties significant for this messaging system. For example, Kafka instrumentations should include partition identifier.
+* `messaging.{system}.destination_original`: Describe the original destination a message (or a batch) were published to.
```

Comment:
I think this is wrong. The other keys are namespace templates. Probably this should be clarified by adding `.*` at the end.

Reply:
I just now realized that destination_original is also a namespace. Then this seems correct, but I 'd still recommend using the format with `.*` at the end.

Reply:
Fixed in [169102d](https://github.com/open-telemetry/semantic-conventions/pull/100/commits/169102d8b648c22d542c97cea5925d16c9a780d1)